src/common/models/problem_sets.ts

```diff
@@ -67,11 +69,11 @@ export class ProblemSet extends Model {
 	}
 
 	set(params: ParseableProblemSet) {
-		if (params.set_id != undefined) this.set_id = params.set_id;
-		if (params.set_visible != undefined) this.set_visible = params.set_visible;
+		if (params.set_id) this.set_id = params.set_id;
+		if (params.set_visible !== undefined) this.set_visible = params.set_visible;
 		if (params.course_id != undefined) this.course_id = params.course_id;
```

Comment:
I've been trying to use truthy checks for string types to prevent setting fields to the empty string '' and otherwise checking != undefined.

Reply:
So, for booleans, we clearly cannot rely on truthy/falsey; and same for any numerical values that _could_ be zero... but beyond that, a truthy check should suffice.

Especially because (type-wise) we are very broad about the types we allow, these undefined checks would permit empty strings...